src/main/java/bio/terra/common/stairway/StairwayComponent.java

```diff
@@ -0,0 +1,138 @@
+package bio.terra.common.stairway;
+
+import static com.google.cloud.ServiceOptions.getDefaultProjectId;
+
+import bio.terra.common.kubernetes.KubeService;
+import bio.terra.common.kubernetes.KubernetesComponent;
+import bio.terra.stairway.Stairway;
+import bio.terra.stairway.exception.StairwayException;
+import bio.terra.stairway.exception.StairwayExecutionException;
+import java.util.LinkedList;
+import java.util.List;
+import java.util.Set;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.atomic.AtomicReference;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.context.ApplicationContext;
+import org.springframework.stereotype.Component;
+
+/** A Spring Component for exposing an initialized {@link Stairway}. */
+@Component
+public class StairwayComponent {
+  private final Logger logger = LoggerFactory.getLogger(StairwayComponent.class);
+
+  private final StairwayConfiguration stairwayConfiguration;
+  private final StairwayJdbcConfiguration stairwayJdbcConfiguration;
+  private final Stairway stairway;
+  private final KubeService kubeService;
+
+  public enum Status {
+    INITIALIZING,
+    OK,
+    ERROR,
+    SHUTDOWN,
+  }
+
+  private final AtomicReference<Status> status = new AtomicReference<>(Status.INITIALIZING);
+
+  @Autowired
+  public StairwayComponent(
+      ApplicationContext applicationContext,
+      StairwayConfiguration stairwayConfiguration,
+      StairwayJdbcConfiguration stairwayJdbcConfiguration,
+      KubernetesComponent kubernetesComponent) {
+    this.stairwayConfiguration = stairwayConfiguration;
+    this.stairwayJdbcConfiguration = stairwayJdbcConfiguration;
+    this.kubeService = kubernetesComponent.get();
+    String stairwayClusterName = kubeService.getNamespace() + "buffer--stairwaycluster";
+    logger.info(
+        "Creating Stairway: name: [{}]  cluster name: [{}]",
+        kubeService.getPodName(),
+        stairwayClusterName);
+
+    // TODO(PF-314): Cleanup old flightlogs.
+    // TODO(PF-456):
+    Stairway.Builder builder =
+        Stairway.newBuilder()
+            .maxParallelFlights(stairwayConfiguration.getMaxParallelFlights())
+            .applicationContext(applicationContext)
+            .keepFlightLog(true)
+            .stairwayName(kubeService.getPodName())
+            .stairwayClusterName(stairwayClusterName)
+            .workQueueProjectId(getDefaultProjectId())
+            .enableWorkQueue(kubernetesComponent.isInKubernetes())
+            .stairwayHook(new TracingHook());
```

Comment:
At least the hook should be customizable, not sure if we need to customize other configs.

Reply:
How do I make that customizable? As a constructor parameter, like a `Provider<TracingHook>`?

Reply:
We may want to pass `List<StairwayHook>`. 
One possible solution is the constructor just build Stairway.Builder. Then change the following method to `initialize(List<StairwayHook> stairwayHooks)`

Reply:
For example: WSM needs two hooks:
https://github.com/DataBiosphere/terra-workspace-manager/blob/dev/src/main/java/bio/terra/workspace/service/job/JobService.java#L86

Reply:
Why not `Provider<List<StairwayHook>>` in the constructor? If I build the builder in the ctor, how do I add the hooks to it in `initialize()`?

Reply:
that SG!